src/core/xconnection.rs

```diff
@@ -585,9 +590,13 @@ impl XConn for XcbConnection {
 
                 xcb::KEY_PRESS => {
                     let e: &xcb::KeyPressEvent = unsafe { xcb::cast_event(&event) };
-                    Some(XEvent::KeyPress {
-                        code: KeyCode::from_key_press(e),
-                    })
+                    let mut code = KeyCode::from_key_press(e);
+
+                    if let Some(&numlock_mask) = self.modifier_map.get("Num_Lock") {
```

Comment:
If we are only using `Num_lock` at this stage, can we just pass that through explicitly rather than grabbing all of the modifiers and keying in to the map every time?